src/App.tsx

```diff
@@ -2,108 +1,75 @@
+import React, { useState } from 'react';
+
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const optionsPerPage = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(optionsPerPage[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const firstVisibleItemIndex = (currentPage - 1) * perPage;
+  const lastVisibleItemIndex = Math.min(items.length, currentPage * perPage);
+
+  const visibleItems = items.slice(firstVisibleItemIndex, lastVisibleItemIndex);
+
+  const handlePageChange = (newPage: number) => {
+    const maxPage = Math.ceil(items.length / perPage);
+
+    if (newPage >= 1 && newPage <= maxPage) {
+      setCurrentPage(newPage);
+    }
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
```

Comment:
When changing the number of items per page, it's important to check if the current page still has items to display. If the new 'perPage' value results in a current page without items, you should calculate and set the appropriate page number that contains items.